zipkin-cassandra/src/main/scala/com/twitter/zipkin/storage/cassandra/CassandraSpanStore.scala

```diff
@@ -292,27 +298,44 @@ abstract class CassandraSpanStore(
   }
 
   override def getTraceIdsByAnnotation(
-    serviceName: String,
+    serviceName: Option[String],
     annotation: String,
     value: Option[ByteBuffer],
     endTs: Long,
     lookback: Long,
     limit: Int
   ): Future[Seq[IndexedTraceId]] = {
     QueryGetTraceIdsByAnnotationCounter.incr()
+    serviceName match {
+      case Some(name) => FutureUtil.toFuture(
+        repository
+          .getTraceIdsByAnnotation(annotationKey(name, annotation, value), endTs * 1000, lookback * 1000, limit))
+        .map { traceIds =>
+          traceIds.asScala
+            .map { case (traceId, ts) => IndexedTraceId(traceId, timestamp = ts) }
+            .toSeq
+        }
+      case None => FutureUtil.toFuture(repository.getServiceNames).flatMap { names =>
+        val futures: Seq[Future[util.Map[java.lang.Long, java.lang.Long]]] = names.asScala.toSeq.map { name =>
+          FutureUtil.toFuture(repository
```

Comment:
this will likely cause some havoc, as there's sometimes thousands of service names. Maybe throw UnsupportedException for now? Then we can open up another issue to add an index for annotation w/o service name